### PR TITLE
Added stratification to train/test split

### LINKED ISSUE #20: stratified k-fold validation
Add code to use a key given in config file to do a stratified k-fold validation, e.g. Layout

### PR DESCRIPTION
**Pull request by jacob-evarts**

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 def main(config):
     cfg = config["cs"]
     sobol_power = config["sobol_power"]
+    stratify = config["stratify"]
```
Comment (cainja) [inline src/config.py:14]: check for KeyError here, if you make it optional later on.

File: src/conf/config.yaml [hunk 2]
```diff
@@ -3,4 +3,5 @@ defaults:
   - _self_
 
 sobol_power: 3
+stratify: LAYOUT
```
Comment (cainja) [inline src/conf/config.yaml:6]: I don't think it is possible to not specify this. Follow up comment below.

**Review by cainja — CHANGES_REQUESTED** [review]

File: src/conf/config.yaml (hunk 2 above)
Comment (jacob-evarts) [reply src/conf/config.yaml:6]: apparently if you leave it blank it just sets it to None, so no need to check for a keyError either

**Review by cainja — APPROVED** [review]
good work

**Pull request merged**